tests/TestCase/Database/Expression/CaseStatementExpressionTest.php

```diff
@@ -218,6 +219,7 @@ public function valueTypeInferenceDataProvider(): array
             [ChronosDate::now(), 'date'],
             [Chronos::now(), 'datetime'],
             [new IdentifierExpression('Table.column'), null],
+            [new FunctionExpression('SUM', ['Table.column' => 'literal'], [], 'integer'), null],
```

Comment:
What is this testing if the expected type is null?

Reply:
That not just identifier expressions are excluded, but expressions in general.

Reply:
Why is datetime inferred but the expression type is not?

Reply:
Can you add a comment above these stating these are excluded so this is less confusing?

Reply:
Because that is a value that is going to be bound.

Reply:
> Can you add a comment above these stating these are excluded so this is less confusing?

Will do.